File: CS/Custom editor/RepositoryItemMyCalcEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Popup;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    [UserRepositoryItem("Register")]
    public class RepositoryItemMyCalcEdit : RepositoryItemCalcEdit
    {
        static RepositoryItemMyCalcEdit()
        {
            Register();
        }
        public RepositoryItemMyCalcEdit() { }

        internal const string EditorName = "MyCalcEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyCalcEdit),
                typeof(RepositoryItemMyCalcEdit), typeof(DevExpress.XtraEditors.ViewInfo.CalcEditViewInfo),
                new ButtonEditPainter(), true, null));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        private int _SecondActionButtonIndex;
        public int SecondActionButtonIndex
        {
            get { return _SecondActionButtonIndex; }
            set { _SecondActionButtonIndex = value; }
        }
    }
}

File: VB/Custom editor/RepositoryItemMyCalcEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Popup

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory(""), UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyCalcEdit
		Inherits RepositoryItemCalcEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Friend Const EditorName As String = "MyCalcEdit"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyCalcEdit), GetType(RepositoryItemMyCalcEdit), GetType(DevExpress.XtraEditors.ViewInfo.CalcEditViewInfo), New ButtonEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

		Private _SecondActionButtonIndex As Integer
		Public Property SecondActionButtonIndex() As Integer
			Get
				Return _SecondActionButtonIndex
			End Get
			Set(ByVal value As Integer)
				_SecondActionButtonIndex = value
			End Set
		End Property
	End Class
End Namespace
